### PR TITLE
*: Fix infoschema/perfschema concurrent bugs

### PR DESCRIPTION
**Pull request by shenli**
Make it safe to create Handle on multiple stores.
Fix sql logic test.

### CODE REVIEW

**Comment by shenli** [comment]
@coocood @zimulala PTAL

File: perfschema/statement.go [hunk 1]
```diff
@@ -97,9 +97,11 @@ type StatementState struct {
 	noGoodIndexUsed uint8
 }
 
+/*
 var (
 	stmtInfos = make(map[reflect.Type]*statementInfo)
 )
+*/
```
Comment (coocood) [inline perfschema/statement.go]: This can be removed?

File: infoschema/infoschema.go [hunk 2]
```diff
@@ -228,27 +228,47 @@ func (is *infoSchema) Clone() (result []*model.DBInfo) {
...
 		store: store,
 	}
 	// init memory tables
-	initMemoryTables()
-	initPerfSchema()
-	return h
+	var err error
+	h.memSchema, err = newMemSchemaHandle()
+	if err != nil {
+		if err != nil {
```
Comment (zimulala) [inline infoschema/infoschema.go]: Is it duplicate ?

**Comment by shenli** [comment]
PTAL @zimulala  @coocood

**Comment by coocood** [comment]
LGTM

File: infoschema/infoschema_test.go [hunk 3]
```diff
@@ -194,6 +197,33 @@ func (*testSuite) TestT(c *C) {
 	c.Assert(col, NotNil)
 }
 
+// Make sure it is safe to concurrently create handle on multiple stores.
+func (testSuite) TestConcurrent(c *C) {
```
Comment (zimulala) [inline infoschema/infoschema_test.go]: Add `defer testleak.AfterTest(c)()` ?

Comment (shenli) [inline infoschema/infoschema_test.go]: @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**